my-dsl.md

```diff
@@ -1,39 +1,54 @@
+
 # DSL-ness
 
 _Fowler writes about a spectrum of languages, from general-purpose languages to
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum,
 and why?_
 
+Regular Languages fall in the spectrum of more "purely" domain-specific becuase it's language is used as a tool within general-purpose languages such as java and python. Regualr languages abide by all of Fowler's definitions of a DSL and fall into the external DSL catagory. 
+
 # Internal or external?
 
 _Is the language implemented as an internal or external DSL?
 Justify your answer._
 
+Regular Languages are an external DSL becuase it is utalized as a tool within other main stream languages such as java and python. Additionally, Regular languages are in themselves a differnt language from the host language in which they are used and require differnt syntax then the host language. 
+
 # Host language
 
 _What language(s) was (were) used to implement the DSL?_
 
+The first implamentaion of regualr expressions was created by Ken Thompson, who implemented the idea of regular expressions inside the text editor called ed. Later in the late 80's, Larry Wall built a Perl programming language that helped regular expressions become mainstream. Now, regualr expressions are supported by modern programming languages, text processing programs and text editors. They can often be found in many of todays Java and Python projects. [Link to History of Regex](https://scantopdf.com/blog/the-history-of-regex/)
+
 # Benefits
 
 _Identify one potential benefit of the DSL: how is a programmer's life made
 easier by the existence of this language?_
 
+Regular expressions make a programmer's life easier by analyzing strings for patterns. AKA pattern recogintion. This makes the programers life easier becuase they do not have to define a program to find a pattern in a string instead they can simply use a Regular expression which will find the pattern in a stringg for them.
+
 # Drawbacks
 
 _Identify one potential drawback of the DSL: what does a programmer
 lose by using this DSL instead of a general-purpose language?_
+
+Regular expressions have alot of potential cost in terms of time. The longer the string the longer Regular expressions take to analyze the text for any particular patterns. Additionally they are quite specific. What I mean by this is that Regular expressions have a hard time understanding special characters, white space, and line breaks. 
```

Comment:
This is a good point about the recursive nature of regex computation, and how it might not always be the fastest. I also think about how much I have struggled with understanding regex in the past, and how when there are weird things like special characters, because (as you've pointed out) regex doesn't handle it well, the programmer might also lose time trying to fight the language to do what they wanted.